packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -253,84 +254,120 @@ export const addPositionPropertiesToSections = <TSection extends FieldSection>(
+
+    sections.push({
+      ...sectionConfig,
+      formatValue: currentTokenValue,
+      value: sectionValue,
+      placeholder: getSectionPlaceholder(utils, localeText, sectionConfig, currentTokenValue),
+      separator: '',
+      edited: false,
+    });
+
+    currentTokenValue = '';
+  };
+
   for (let i = 0; i < expandedFormat.length; i += 1) {
     const char = expandedFormat[i];
-    if (!char.match(/([A-zÀ-ú]+)/g)) {
-      if (currentTokenValue === '') {
-        sections[sections.length - 1].separator += char;
-      } else {
-        const dateForCurrentToken = formatDateWithPlaceholder(utils, date, currentTokenValue);
-        if (dateForCurrentToken === currentTokenValue) {
-          sections[sections.length - 1].separator += currentTokenValue;
-          currentTokenValue = '';
-        } else {
-          const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
-
-          sections.push({
-            ...sectionConfig,
-            formatValue: currentTokenValue,
-            value: dateForCurrentToken,
-            emptyValue: sectionConfig.dateSectionName,
-            separator: char,
-            edited: false,
-          });
-          currentTokenValue = '';
-        }
-      }
-    } else {
-      currentTokenValue += char;
-    }
 
-    if (i === expandedFormat.length - 1) {
-      const dateForCurrentToken = formatDateWithPlaceholder(utils, date, currentTokenValue);
-      if (dateForCurrentToken === currentTokenValue) {
-        sections[sections.length - 1].separator += currentTokenValue;
-      } else {
-        const sectionConfig = getDateSectionConfigFromFormatToken(utils, currentTokenValue);
-
-        sections.push({
-          ...sectionConfig,
-          formatValue: currentTokenValue,
-          value: dateForCurrentToken,
-          emptyValue: sectionConfig.dateSectionName,
-          separator: null,
-          edited: false,
-        });
-      }
+    if (char.match(/([A-zÀ-ú]+)/g)) {
+      currentTokenValue += char;
+    } else {
+      commitCurrentToken();
+      sections[sections.length - 1].separator += char;
     }
   }
 
-  return sections;
+  commitCurrentToken();
+
+  return sections.map((section) => {
+    if (section.separator !== '/') {
+      return section;
+    }
+    return {
+      ...section,
+      separator: ' / ',
+      parsingSeparator: '/',
+    };
+  });
 };
 
-export const createDateStrFromSections = (sections: FieldSection[]) =>
+export const createDateStrFromSections = (
+  sections: FieldSection[],
+  willBeRenderedInInput: boolean,
+) =>
   sections
     .map((section) => {
       let sectionValueStr = getSectionVisibleValue(section);
 
-      if (section.separator != null) {
-        sectionValueStr += section.separator;
+      const separator = willBeRenderedInInput
+        ? section.separator
+        : section.parsingSeparator ?? section.separator;
```

Comment:
It would not work with the `start` and `end` property
I think that was why I did not choose that option.

Reply:
A last idea to avoid those double property could be to modify that to the root as follow

```js
const expandedFormat = utils.expandFormat(format).replace(/\s?\/\s?/g, ' / ');
```

Reply:
Same problem:

```tsx
const newDate = utils.parse(createDateStrFromSections(activeDateSections, false), format);
```

When you parse the string created from the sections, you will have ` / ` instead of `/` and the date library can refuse to parse.
Unless we also store the expanded format in state instead of using the one from the prop.
But not sure it's a good trade-off

Reply:
And technically we can't do `utils.parse(createDateStrFromSections(activeDateSections).replace(/ \/ /g, '/'), format);`
Because the format could actually have ` / ` :laughing:

Reply:
This would be the same behavior as what is currently done no? 🤔

Reply:
No
Imagine you pass the format `format="DD / MM"`
If we do remove the space across the `/` before parsing, then we would parse with the format `DD/MM`, which the date library might not understand depending on how strict they are.

Reply:
Ok, I just get the point of your message about
```js
const newDate = utils.parse(createDateStrFromSections(activeDateSections), format);
```

I thought that `format` prop was only provided by the dev/default props. But internally we only use `expandedFormat` to be sure that the all format and the sections formats are coherent